src/Files.App/UserControls/NavigationToolbar.xaml.cs

```diff
@@ -195,68 +190,125 @@ private async Task AddHistoryItemsAsync(IEnumerable<PageStackEntry> items, IList
+						}
+
+						// Navigate forward
+						shellPage.Forward_Click();
+					}
+				}
 			}
 		}
 
-		private async Task LoadFlyoutItemIconAsync(MenuFlyoutItem flyoutItem, string path)
+		private void ClickablePath_GettingFocus(UIElement sender, GettingFocusEventArgs args)
 		{
-			var imageSource = await NavigationHelpers.GetIconForPathAsync(path);
+			if (args.InputDevice != FocusInputDeviceKind.Keyboard)
+				return;
 
-			if (imageSource is not null)
-				flyoutItem.Icon = new ImageIcon { Source = imageSource };
+			var previousControl = args.OldFocusedElement as FrameworkElement;
+			if (previousControl?.Name == nameof(HomeButton) || previousControl?.Name == nameof(Refresh))
+				ViewModel.IsEditModeEnabled = true;
 		}
 
-		private void HistoryItemClicked(ToolbarHistoryItemModel? itemModel)
+		private async void Omnibar_QuerySubmitted(Omnibar sender, OmnibarQuerySubmittedEventArgs args)
 		{
-			if (itemModel is null)
-				return;
+			await ViewModel.HandleItemNavigationAsync(args.Text);
+		}
 
-			var shellPage = Ioc.Default.GetRequiredService<IContentPageContext>().ShellPage;
-			if (shellPage is null)
-				return;
+		private async void Omnibar_SuggestionChosen(Omnibar sender, OmnibarSuggestionChosenEventArgs args)
+		{
+			if (args.SelectedItem is OmnibarPathModeSuggestionModel item &&
+				!string.IsNullOrEmpty(item.Path))
+				await ViewModel.HandleItemNavigationAsync(item.Path);
+		}
 
-			if (itemModel.IsBackMode)
-			{
-				// Remove all entries after the target entry in the BackwardStack
-				while (shellPage.BackwardStack.Last() != itemModel.PageStackEntry)
-				{
-					shellPage.BackwardStack.RemoveAt(shellPage.BackwardStack.Count - 1);
-				}
+		private async void Omnibar_TextChanged(Omnibar sender, OmnibarTextChangedEventArgs args)
+		{
+			await ViewModel.PopulateOmnibarSuggestionsForPathMode();
+		}
 
-				// Navigate back
-				shellPage.Back_Click();
-			}
-			else
+		private async void BreadcrumbBar_ItemClicked(Controls.BreadcrumbBar sender, Controls.BreadcrumbBarItemClickedEventArgs args)
+		{
+			if (args.IsRootItem)
 			{
-				// Remove all entries before the target entry in the ForwardStack
-				while (shellPage.ForwardStack.First() != itemModel.PageStackEntry)
-				{
-					shellPage.ForwardStack.RemoveAt(0);
-				}
-
-				// Navigate forward
-				shellPage.Forward_Click();
+				await ViewModel.HandleItemNavigationAsync("Home");
+				return;
 			}
+
+			await ViewModel.HandleFolderNavigationAsync(ViewModel.PathComponents[args.Index].Path);
 		}
 
-		private void ClickablePath_GettingFocus(UIElement sender, GettingFocusEventArgs args)
+		private async void BreadcrumbBar_ItemDropDownFlyoutOpening(object sender, BreadcrumbBarItemDropDownFlyoutEventArgs e)
 		{
-			if (args.InputDevice != FocusInputDeviceKind.Keyboard)
+			if (e.IsRootItem)
+			{
+				// TODO: Populate a different flyout for the root item
+				e.Flyout.Items.Add(new MenuFlyoutHeaderItem() { Text = "Quick access" });
```

Comment:
Can you remove the placeholder menu? I would prefer not merging placeholder code.